### PR TITLE
Add PostgreSQL 11

### PR DESCRIPTION
**Pull request by chtitux**
Difference with a Debian configuration file (generated with `diff /etc/postgresql/11/main/postgresql.conf postgresql.11.conf.j2`)

~~~diff
0a1
> # {{ ansible_managed }}
41c42
< data_directory = '/var/lib/postgresql/11/main'		# use data in another directory
---
> data_directory = '/var/lib/postgresql/{{ postgres_version }}/{{ postgres_cluster_name }}'		# use data in another directory
43c44
< hba_file = '/etc/postgresql/11/main/pg_hba.conf'	# host-based authentication file
---
> hba_file = '/etc/postgresql/{{ postgres_version }}/{{ postgres_cluster_name }}/pg_hba.conf'	# host-based authentication file
45c46
< ident_file = '/etc/postgresql/11/main/pg_ident.conf'	# ident configuration file
---
> ident_file = '/etc/postgresql/{{ postgres_version }}/{{ postgres_cluster_name }}/pg_ident.conf'	# ident configuration file
49c50
< external_pid_file = '/var/run/postgresql/11-main.pid'			# write an extra PID file
---
> external_pid_file = '/var/run/postgresql/{{ postgres_version }}-{{ postgres_cluster_name }}.pid'		# write an extra PID file
59c60
< #listen_addresses = 'localhost'		# what IP address(es) to listen on;
---
> listen_addresses = '{{ postgres_listen_addresses | reject("none") | join(",") }}' # what IP address(es) to listen on;
63,64c64,65
< port = 5436				# (change requires restart)
< max_connections = 100			# (change requires restart)
---
> port = {{ postgres_port }}				# (change requires restart)
> max_connections = 400			# (change requires restart)
88,89c89,90
< #authentication_timeout = 1min		# 1s-600s
< #password_encryption = md5		# md5 or scram-sha-256
---
> #authentication_timeout = 1min          # 1s-600s
> #password_encryption = md5              # md5 or scram-sha-256
100c101
< ssl_cert_file = '/etc/ssl/certs/ssl-cert-snakeoil.pem'
---
> #ssl_cert_file = '/etc/ssl/certs/ssl-cert-snakeoil.pem'
102c103
< ssl_key_file = '/etc/ssl/private/ssl-cert-snakeoil.key'
---
> #ssl_key_file = '/etc/ssl/private/ssl-cert-snakeoil.key'
110d110
< 
117c117,121
< shared_buffers = 128MB			# min 128kB
---
> {% if ansible_memtotal_mb > 65536 %}
> shared_buffers = 16GB			# min 128kB
> {% else %}
> shared_buffers = {{ ansible_memtotal_mb // 4 }}MB			# min 128kB
> {% endif %}
121c125
< #temp_buffers = 8MB			# min 800kB
---
> temp_buffers = 24MB			# min 800kB
126,127c130,131
< #work_mem = 4MB				# min 64kB
< #maintenance_work_mem = 64MB		# min 1MB
---
> work_mem = 512MB				# min 64kB
> maintenance_work_mem = 1024MB		# min 1MB
147a152
> #shared_preload_libraries = ''		# (change requires restart)
159c164
< #bgwriter_delay = 200ms			# 10-10000ms between rounds
---
> bgwriter_delay = 2000ms			# 10-10000ms between rounds
184c189
< #wal_level = replica			# minimal, replica, or logical
---
> wal_level = logical			# minimal, replica, or logical
185a191
> {% if postgres_fsync_enabled %}
190c196,200
< 					# off, local, remote_write, remote_apply, or on
---
> # off, local, remote_write, remote_apply, or on
> {% else %}
> fsync = off				# turns forced synchronization on or off
> synchronous_commit = off		# synchronization level;
> {% endif %}
200c210
< #wal_log_hints = off			# also do full page writes of non-critical updates
---
> wal_log_hints = on			# also do full page writes of non-critical updates
220a231,238
> {% if postgres_archive_enabled %}
> archive_mode = on
> archive_command = 'rsync -a %p barman@{{ postgres_barman_server }}:/var/lib/barman/{{ barman_directory }}/incoming/%f'
> {% else %}
> archive_mode = off
> archive_command = ''
> {% endif %}
> 
239c257
< #max_wal_senders = 10		# max number of walsender processes
---
> max_wal_senders = 10		# max number of walsender processes
241c259
< #wal_keep_segments = 0		# in logfile segments; 0 disables
---
> wal_keep_segments = 16		# in logfile segments; 0 disables
244c262,266
< #max_replication_slots = 10	# max number of replication slots
---
> {% if postgres_max_replication_slots is defined %}
> max_replication_slots = {{ postgres_max_replication_slots }} # max number of replication slots
> {% else %}
> #max_replication_slots = 0	# max number of replication slots
> {% endif %}
263c285
< #hot_standby = on			# "off" disallows queries during recovery
---
> hot_standby = on			# "off" disallows queries during recovery
268c290
< #max_standby_streaming_delay = 30s	# max delay before canceling queries
---
> max_standby_streaming_delay = {{ postgres_extra_config.max_standby_streaming_delay | default('600s') }}	# max delay before canceling queries
315a338,342
> {% if postgres_specific_random_page_cost is defined -%}
> # Random page cost is an arbitrary value relative to the seq_page_cost conf (default 1.0)
> # On SSD disk it is better to lower the default value from 4.0 to 1.0 for instance.
> random_page_cost = {{ postgres_specific_random_page_cost }}			# same scale as above
> {% else -%}
316a344
> {% endif %}
334a363
> effective_cache_size = {{ ansible_memtotal_mb // 2 }}MB
357d385
< 
447c475
< #log_min_duration_statement = -1	# -1 is disabled, 0 logs all statements
---
> log_min_duration_statement = 10000	# -1 is disabled, 0 logs all statements
465c493,495
< log_line_prefix = '%m [%p] %q%u@%d '		# special values:
---
> 
> log_line_prefix = '{{ postgres_log_line_prefix }}'
> #log_line_prefix = '%m [%p] %q%u@%d '	# special values:
492c522
< log_timezone = 'localtime'
---
> log_timezone = 'UTC'
498c528
< cluster_name = '11/main'			# added to process titles if nonempty
---
> cluster_name = '{{ postgres_version }}/{{ postgres_cluster_name }}'			# added to process titles if nonempty
514c544
< stats_temp_directory = '/var/run/postgresql/11-main.pg_stat_tmp'
---
> stats_temp_directory = '/var/run/postgresql/{{ postgres_version }}-{{ postgres_cluster_name }}.pg_stat_tmp'
591c621
< datestyle = 'iso, dmy'
---
> datestyle = '{{ postgres_datestyle }}'
593c623,626
< timezone = 'localtime'
---
> #timezone = '(defaults to server environment setting)'
> {% if postgres_timezone is defined %}
> timezone = '{{ postgres_timezone }}'
> {% endif %}
606c639
< lc_messages = 'en_US.UTF-8'			# locale for system error message
---
> lc_messages = '{{ postgres_locale_system }}'	# locale for system error message
608,610c641,643
< lc_monetary = 'fr_FR.UTF-8'			# locale for monetary formatting
< lc_numeric = 'fr_FR.UTF-8'			# locale for number formatting
< lc_time = 'fr_FR.UTF-8'				# locale for time formatting
---
> lc_monetary = '{{ postgres_locale_formats }}'	# locale for monetary formatting
> lc_numeric = '{{ postgres_locale_formats }}'	# locale for number formatting
> lc_time = '{{ postgres_locale_formats }}'	# locale for time formatting
613c646
< default_text_search_config = 'pg_catalog.english'
---
> default_text_search_config = '{{ postgres_text_search_config }}'
617c650
< #shared_preload_libraries = ''	# (change requires restart)
---
> #shared_preload_libraries = ''		# (change requires restart)
620c653
< #jit_provider = 'llvmjit'		# JIT library to use
---
> #jit_provider = 'llvmjit'               # JIT library to use
678c711
< include_dir = 'conf.d'			# include files ending in '.conf' from
---
> #include_dir = 'conf.d'			# include files ending in '.conf' from
~~~

### CODE REVIEW

File: templates/postgresql.11.conf.j2 [hunk 1]
```diff
@@ -0,0 +1,721 @@
...
+# - Disk -
+
+#temp_file_limit = -1			# limits per-process temp file space
+					# in kB, or -1 for no limit
+
+# - Kernel Resources -
+
+#max_files_per_process = 1000		# min 25
+					# (change requires restart)
+#shared_preload_libraries = ''		# (change requires restart)
```
Comment (paulRbr) [inline templates/postgresql.11.conf.j2]: This line is already present later in the configuration file (and in the good section)

File: templates/postgresql.11.conf.j2 [hunk 2]
```diff
@@ -0,0 +1,721 @@
...
+# - Settings -
+
+wal_level = logical			# minimal, replica, or logical
+					# (change requires restart)
+{% if postgres_fsync_enabled %}
+#fsync = on				# flush data to disk for crash safety
+					# (turning this off can cause
+					# unrecoverable data corruption)
+#synchronous_commit = on		# synchronization level;
+# off, local, remote_write, remote_apply, or on
```
Comment (paulRbr) [inline templates/postgresql.11.conf.j2]:
```suggestion
 					# off, local, remote_write, remote_apply, or on
```
Missing spacing

File: templates/postgresql.11.conf.j2 [hunk 3]
```diff
@@ -0,0 +1,721 @@
...
+
+max_wal_senders = 10		# max number of walsender processes
+				# (change requires restart)
+wal_keep_segments = 16		# in logfile segments; 0 disables
+#wal_sender_timeout = 60s	# in milliseconds; 0 disables
+
+{% if postgres_max_replication_slots is defined %}
+max_replication_slots = {{ postgres_max_replication_slots }} # max number of replication slots
+{% else %}
+#max_replication_slots = 0	# max number of replication slots
```
Comment (paulRbr) [inline templates/postgresql.11.conf.j2]:
```suggestion
#max_replication_slots = 10	# max number of replication slots
```
It seems the default comment in the Debian package you have mentioned in the description of your PR is at 10 not 0

**Review by paulRbr — CHANGES_REQUESTED** [review]
Thanks! LGTM overall. I left a few comments that needs changes.

File: templates/postgresql.11.conf.j2 (hunk 1 above)
Comment (chtitux) [reply templates/postgresql.11.conf.j2]: indeed, removed

File: templates/postgresql.11.conf.j2 (hunk 3 above)
Comment (chtitux) [reply templates/postgresql.11.conf.j2]: indeed, I put 10 there

File: templates/postgresql.11.conf.j2 (hunk 2 above)
Comment (chtitux) [reply templates/postgresql.11.conf.j2]: :+1:

**Review by paulRbr — APPROVED** [review]
Thanks for the changes!

**Pull request merged**